app/controllers/application_controller.rb

```diff
@@ -185,7 +198,32 @@ def bad_request e
 
   def unauthorized e
     logger.debug "#{e}\n#{e.backtrace.join "\n"}"
-    head :unauthorized
+    if e.message == Authentication::AuthnOidc::IdTokenExpired.new.to_s
```

Comment:
this is very specific. what if we decide that also for another scenario we want to return the error message too? we'll need to update both `authenticate_controller.rb` and this file. please find a way that doesn't include the message check.

a possible solution is to check that the message is not "ApplicationController::Unauthorized" which is the default value (and thus if we want some exception to return the message to the user we raise it with message in `authenticate_controller.rb` as we do for `Authentication::AuthnOidc::IdTokenExpired`.

please check if there is a neater solution. maybe enhance this class in `application_controller.rb`
```
  class Unauthorized < RuntimeError
  end
```